common/src/main/java/io/github/consistencyplus/consistency_plus/core/entries/block/AbstractRegistryEntry.java

```diff
@@ -0,0 +1,74 @@
+package io.github.consistencyplus.consistency_plus.core.entries.block;
+
+
+import dev.architectury.registry.registries.RegistrySupplier;
+import io.github.consistencyplus.consistency_plus.base.ConsistencyPlusMain;
+import io.github.consistencyplus.consistency_plus.blocks.BlockTypes;
+import io.github.consistencyplus.consistency_plus.blocks.BlockShapes;
+import io.github.consistencyplus.consistency_plus.core.entries.interfaces.BlockRegistryInterface;
+import io.github.consistencyplus.consistency_plus.core.extensions.CPlusFenceGateBlock;
+import io.github.consistencyplus.consistency_plus.core.extensions.CPlusStairBlock;
+import io.github.consistencyplus.consistency_plus.registry.CPlusEnhancedRegistry;
+import net.minecraft.block.*;
+import net.minecraft.item.Item;
+
+import static io.github.consistencyplus.consistency_plus.registry.CPlusEnhancedRegistry.checkMinecraft;
+
+public abstract class AbstractRegistryEntry implements BlockRegistryInterface {
+    // This note is more towards Jay than anyone else.
+    // For now, this is the solution Im choosing, however, after 0.5, we can change the way these are registered.
+    public static String name;
+    public static AbstractBlock.Settings blockSettings;
+
+    public AbstractRegistryEntry(String name, AbstractBlock.Settings blockSettings) {
+        AbstractRegistryEntry.name = name;
+        AbstractRegistryEntry.blockSettings = blockSettings;
+        construct();
+    }
+
+    public AbstractRegistryEntry(String name, AbstractBlock.Settings blockSettings, boolean construct) {
+        //Use this as a way to reference the code without constructing the actual blocks. Useful for DyedRegistryEntry
+        AbstractRegistryEntry.name = name;
+        AbstractRegistryEntry.blockSettings = blockSettings;
+        if (construct) construct();
+    }
+
+    public void construct() {
```

Comment:
there is a lot of code duplication in overriding methods that probably isn't needed.

Reply:
The best way I think it could be done is separating them into a third check method thats actually able to implement exclusive code for that set.